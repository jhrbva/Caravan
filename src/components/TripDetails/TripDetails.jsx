import React from 'react';
import { Link } from 'react-router-dom';
import { Button } from 'react-bootstrap';

import './TripDetails.scss';

export const getDateTime = (date) => {
	const tripDate = date.slice(0, 10);
	const tripTime = date.slice(11, 19);
	return { tripDate, tripTime };
};

export const getRestStops = (reststops) => {
	if (reststops.length > 0) {
		return (
			<div>
				<span className='trip-details-headings'>Stoping at</span>
				<ul>
					<li>Stop 1</li>
				</ul>
			</div>
		);
	}
	console.log(reststops);
};

const TripDetails = (props) => {
	const {
		trip_title,
		tripdate,
		trip_description,
		startlocation,
		destination,
	} = props.trip;

	const { reststops } = props;

	const { tripDate, tripTime } = getDateTime(tripdate);

	return (
		<div>
			<div className='trip-details-wrapper'>
				<div className='trip-details-header'>
					<h1 className='header-text'>{trip_title}</h1>
					<h2 className='header-text-light'>{trip_description}</h2>
				</div>
				<div className='trip-details'>
					<p>
						<span className='trip-details-headings'>You were invited by </span>@
						Host
					</p>
					<p>
						<span className='trip-details-headings'>When:</span> {tripDate}
					</p>
					<p>
						<span className='trip-details-headings'>Leaving at:</span>
						{tripTime}
					</p>
					<p>
						<span className='trip-details-headings'>From</span>
						<br /> {startlocation}
					</p>
					<p>
						<span className='trip-details-headings'>To</span>
						<br /> {destination}
					</p>

					{getRestStops(reststops)}

					<p>
						<span className='trip-details-headings'>Guest list</span>
					</p>
					<ul>
						<li>@guestA</li>
						<li>@guestB</li>
						<li>@guestC</li>
					</ul>

					<Link to='/map'>
						<Button variant='success'>Start Trip</Button>
					</Link>
				</div>
			</div>
		</div>
	);
};

export default TripDetails;
