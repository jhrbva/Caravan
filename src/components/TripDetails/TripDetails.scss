@import '../../utilities/colorPalette.scss';
@import '../../utilities/mediaQueries.scss';

.trip-details-header {
	padding: 20px;
	text-align: center;
}
.trip-details {
	background: $dream-blue;
	color: white;

	padding: 30px 20px;
}
.trip-details-headings {
	font-weight: bold;
	font-size: 18px;
}
