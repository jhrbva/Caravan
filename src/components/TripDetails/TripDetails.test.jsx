import React from 'react';
import { shallow } from 'enzyme';
import TripDetails, { getDateTime } from './TripDetails';

const testDateTime = '2020-06-01T13:00:06.000Z';
const trip = {
	trip_title: 'Wonders of the Canyon',
	tripdate: testDateTime,
	trip_description: 'Join in as we travel to the Grand Canyon',
	startlocation: '123 OneTreeHill St.',
	destination: 'Grand Canyon, Grand Canyon Ave.',
};
const reststops = [{ location: 'Wawa' }, { location: 'Loves' }];
const host = [
	{
		userid: 1,
		firstname: 'Connie',
		lastname: 'Pink',
		username: 'conniecode',
		email: 'connie@gmail.com',
		phonenumber: '1234567890',
	},
];
const members = [
	[
		{
			userid: 4,
			firstname: 'Julia',
			lastname: 'Red',
			username: 'jcode',
			email: 'julia@gmail.com',
			phonenumber: '4567890123',
		},
	],
];

describe('Trip Details', () => {
	it('should match the snapshot', () => {
		const wrapper = shallow(
			<TripDetails
				trip={trip}
				host={host}
				members={members}
				reststops={reststops}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
