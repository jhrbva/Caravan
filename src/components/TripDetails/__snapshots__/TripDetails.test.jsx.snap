// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trip Details should match the snapshot 1`] = `
<div>
  <div
    className="trip-details-wrapper"
  >
    <div
      className="trip-details-header"
    >
      <h1
        className="header-text"
      >
        Wonders of the Canyon
      </h1>
      <h2
        className="header-text-light"
      >
        Join in as we travel to the Grand Canyon
      </h2>
    </div>
    <div
      className="trip-details"
    >
      <p>
        <span
          className="trip-details-headings"
        >
          You were invited by 
        </span>
        @ Host
      </p>
      <p>
        <span
          className="trip-details-headings"
        >
          When:
        </span>
         
        2020-06-01
      </p>
      <p>
        <span
          className="trip-details-headings"
        >
          Leaving at:
        </span>
        13:55:06
      </p>
      <p>
        <span
          className="trip-details-headings"
        >
          From
        </span>
        <br />
         
        123 OneTreeHill St.
      </p>
      <p>
        <span
          className="trip-details-headings"
        >
          To
        </span>
        <br />
         
        Grand Canyon, Grand Canyon Ave.
      </p>
      <div>
        <span
          className="trip-details-headings"
        >
          Stoping at
        </span>
        <ul>
          <li
            key="0"
          >
            Wawa
          </li>
          <li
            key="1"
          >
            Loves
          </li>
        </ul>
      </div>
      <p>
        <span
          className="trip-details-headings"
        >
          Guest list
        </span>
      </p>
      <ul>
        <li>
          @guestA
        </li>
        <li>
          @guestB
        </li>
        <li>
          @guestC
        </li>
      </ul>
      <Link
        to="/map"
      >
        <Button
          active={false}
          disabled={false}
          type="button"
          variant="success"
        >
          Start Trip
        </Button>
      </Link>
    </div>
  </div>
</div>
`;
